### ISSUE TITLE
How do I enter free text in a TID 1501?

### ISSUE THREAD

**Issue by seandoyle** [ImagingDataCommons/highdicom#124]
In row 12 of [TID 1501](http://dicom.nema.org/medical/dicom/current/output/chtml/part16/chapter_A.html#sect_TID_1501) it appears that text entries can be added to the object; the Content Item Description for this line states `Allows encoding a flat list of name-value pairs that are coded questions with coded or text answers, for example, to record categorical observations related to the subject of the measurement group. A single level of coded modifiers may be present.`. What is the recommended way to add these values?

If I create a MeasurementsAndQualitativeEvaluations object containing QualitativeEvaluations of the sort
```
    text_item = QualitativeEvaluation(
        name=CodedConcept(
            value='121071',
            meaning='Finding',
            scheme_designator='DCM',
        ),
        value=CodedConcept(
            value=text,
            meaning='Finding',
            scheme_designator='DCM',
        )
    )
```

this will work (and pass SR validation) until the size of the text exceeds 16 characters (after that there is an exception in coding.py). This may also be considered an abuse of `CodedConcept`?  Or perhaps there is a way to include a`TextContentItem` but I don't see the way to do this.

The use case is to permit free text additions to an SR by a radiologist for findings not present in the original SR generated by the model output. These text findings are at coded anatomic finding sites.

**Comment by hackermd** [comment]
Yes, @CPBridge is right. I wanted to strongly discourage the use of free text for qualitative evaluations.

**Comment by seandoyle** [comment]
Thanks - I agree in general with discouraging free text but the app that I'm creating permits radiologists to enter text (this is their requirement) and don't want to generate a second data object for capturing their output. I'll try @CPBridge's approach and validate it with the pixelmed SR validator.

**Comment by hackermd** [comment]
> Using codes (private or public) assumes that you know what you expect to find in advance. Using private codes just eliminates the burden of finding a standard code. How would you, for example, solve the issue of describing incidental findings using private codes?

I have only really considered automated systems, which need to choose terms from a predefined terminology. This may be different for user interfaces, which need to handle arbitrary user input.

**Comment by hackermd** [comment]
> Implementing this in a backwards compatible way could get a bit messy. We'd probably need to create a new, alternative `TextQualitativeEvaluation`, or similar, and then accept `Sequence[Union[QualitativeEvaluation, TextQualitativeEvaluation]]` for the `qualitative_evaluations` parameter of the measurement group constructors. It's also true that currently the parsing API won't return content items of value type TEXT for the `get_qualitative_evaluations` method

We could call those TEXT content items "descriptions" rather than "evaluations" and add a `descriptions` parameter to the constructor as well as a corresponding `get_descriptions` method to the class.
